### PR TITLE
Upgrade modules to Decred v1.5.0 release

### PR DESCRIPTION
**Pull request by C-ollins**
All of dcrd & dcrwallet modules were bumped to the module versions used in Decred v1.5.0 release, and the library changes were also implemented in affected files. The modules in the current release are outdated. An `IsRescanning` function was added to enable the frontend to determine what rescan notification is coming from the sync interface.

### CODE REVIEW

File: dcrlibwallet.go [hunk 1]
```diff
@@ -653,22 +622,27 @@ func (lw *LibWallet) RescanBlocks() error {
 		default:
 			for _, response := range lw.syncResponses {
 				response.OnRescan(totalHeight, FINISH)
+				response.OnSynced(true)
```
[the remark this answers is not in the document]
Comment (C-ollins) [reply dcrlibwallet.go]: The frontend has different callbacks listening for `OnSynced` and calling `OnSynced` would enable the frontend remove the rescan progress bar and set up the interface for a synced wallet.

[the remark this answers is not in the document]
Comment (itswisdomagain) [reply dcrlibwallet.go]: What if the frontend uses the `FINISH` state together with `IsRescanning()` to determine that the rescan has completed and then locally call the `onSynced()` function? This added bit here could cause issues for dcrios.

File: dcrlibwallet.go [hunk 2]
```diff
@@ -560,14 +518,30 @@ func (lw *LibWallet) RpcSync(networkAddress string, username string, password st
...
+
+	rpcSyncer.SetCallbacks(ntfns)
 
 	go func() {
 		// Run wallet synchronization until it is cancelled or errors.  If the
 		// context was cancelled, return immediately instead of trying to
 		// reconnect.
-		err := syncer.Run(ctx, true)
+		ctx, cancel := contextWithShutdownCancel()
+		lw.cancelSync = cancel
```
Comment (itswisdomagain) [inline dcrlibwallet.go]: Line 419 in the `SpvSync()` method can benefit from this change as well.

File: dcrlibwallet.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (C-ollins) [reply dcrlibwallet.go]: Dcrandroid uses same sync UI for rescan which gets cleaned up in onSynced. This version of dcrlibwallet is not compatible with any dcrios release.

[the remark this answers is not in the document]
Comment (itswisdomagain) [reply dcrlibwallet.go]: Noted.

**Pull request merged**